@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -623,6 +706,7 @@ function placePointLabelAtCurrentAnchor(
     // present in the new zoom level) before an even lower priority one takes the space.
     // Otherwise the lowest priority text will fade in and back out.
     // TODO: Add a unit test for this scenario.
+    // TODO!: Move this out of the function to make it full utility only!
```

Comment:
Why do you want to move this out. Placement has two main responsibilities: find free screen space, and allocate it so that other labels can't use it. This code here is the second, so it belongs here.

Reply:
I left this comment here to make a note that this function will be soon changed to simple utility that will only CHECK not PERFORM the placement, I would like to get rid of TextCanvas.measureText most of the time for alternative placement.
The idea is to perform it once! Then text bounds may be computed based on **base placement**, **base bounds** that will be computed only once for entire label live, and the **anchor point** difference between the base and current.

Reply:
Almost forgot to mention with that change in mind I will get rid of reverting back out screen position, bounds etc... Utility will simply work on the cache objects and if placement will be found we will copy from cache to the output variables.